# HG changeset patch
# User Rik <code@nomad.inbox5.com>
# Date 1271972338 25200
#      Thu Apr 22 14:38:58 2010 -0700
# Node ID e9d1a974a517f2bb7ac02764d7f6174c94e210b6
# Parent  952d4df5b68673a624fd44dc573aaf3d424a63bf
Eliminate repmat (Inf, x , y) and simply use Inf (x,y)

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-04-22  Rik <octave@nomad.inbox5.com>
+
+	* optimization/glpk.m: Eliminate repmat (Inf, x , y) and just use Inf (x,y)
+
 2010-04-22  Rik <octave@nomad.inbox5.com>
 
 	* general/rat.m, sparse/gplot.m, strings/base2dec.m, geometry/triplot.m
      optimization/sqp.m, optimization/qp.m, statistics/base/run_count.m,
      statistics/base/ranks.m: Eliminate NaN*ones and Inf*ones constructs
      and just use Nan() and Inf()
 
 2010-04-19  Ben Abbott <bpabbott@mac.com>
diff --git a/scripts/optimization/glpk.m b/scripts/optimization/glpk.m
--- a/scripts/optimization/glpk.m
+++ b/scripts/optimization/glpk.m
@@ -470,23 +470,23 @@ function [xopt, fmin, status, extra] = g
   else
     lb = zeros (nx, 1);
   endif
 
   ## 5) Vector with the upper bound of each variable
 
   if (nargin > 4)
     if (isempty (ub))
-      ub = repmat (Inf, nx, 1);
+      ub = Inf (nx, 1);
     elseif (! isreal (ub) || all (size (ub) > 1) || length (ub) != nx)
       error ("UB must be a real valued %d by 1 column vector", nx);
       return;
     endif
   else
-    ub = repmat (Inf, nx, 1);
+    ub = Inf (nx, 1);
   endif
 
   ## 6) Sense of each constraint
 
   if (nargin > 5)
     if (isempty (ctype))
       ctype = repmat ("S", nc, 1);
     elseif (! ischar (ctype) || all (size (ctype) > 1) || length (ctype) != nc)
